src/plugins/gs-plugin-external-apps.c

```diff
@@ -509,6 +563,12 @@ extract_runtime_info_from_json_data (const char *data,
 		return NULL;
 	}
 
+	/* optional elements */
+	type_str = json_object_get_string_member (runtime, JSON_RUNTIME_TYPE_KEY);
+	if (type_str) {
```

Comment:
`g_strdup()` works just fine with `NULL`, so I think you don't need to check for this here.